src/common/buffer.cc

```diff
@@ -45,7 +45,7 @@ using namespace ceph;
 #define CEPH_BUFFER_APPEND_SIZE (CEPH_BUFFER_ALLOC_UNIT - sizeof(raw_combined))
 
 #ifdef BUFFER_DEBUG
-# define bdout { std::lock_guard lg(ceph::spinlock()); std::cout
+# define bdout { std::lock_guard<ceph::spinlock> lg(ceph::spinlock()); std::cout
```

Comment:
this does not look right. how are the different threads supposed to wait for each other for accessing `std::count`? as they are not waiting on the same `spinlock` instance.